packages/core/useBluetooth/index.ts

```diff
@@ -0,0 +1,132 @@
+import type { ConfigurableNavigator } from '../_configurable'
+
+import { defaultNavigator } from '../_configurable'
+
+export interface UseBluetoothRequestDeviceOptions {
+  /**
+   *
+   * An array of BluetoothScanFilters. This filter consists of an array
+   * of BluetoothServiceUUIDs, a name parameter, and a namePrefix parameter.
+   *
+   */
+  filters?: BluetoothLEScanFilter[] | undefined
+  /**
+   *
+   * An array of BluetoothServiceUUIDs.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/BluetoothRemoteGATTService/uuid
+   *
+   */
+  optionalServices?: BluetoothServiceUUID[] | undefined
+}
+
+export interface UseBluetoothOptions extends UseBluetoothRequestDeviceOptions, ConfigurableNavigator {
+  /**
+   *
+   * A boolean value indicating that the requesting script can accept all Bluetooth
+   * devices. The default is false.
+   *
+   * !! This may result in a bunch of unrelated devices being shown
+   * in the chooser and energy being wasted as there are no filters.
+   *
+   *
+   * Use it with caution.
+   *
+   * @default false
+   *
+   */
+  acceptAllDevices?: boolean
+}
+
+export function useBluetooth(options?: UseBluetoothOptions) {
+  let {
+    acceptAllDevices = false,
+  } = options || {}
+
+  const {
+    filters = undefined,
+    optionalServices = undefined,
+    navigator = defaultNavigator,
+  } = options || {}
+
+  const isSupported = navigator && 'bluetooth' in navigator
+
+  const device = ref<undefined | BluetoothDevice>(undefined)
+
+  const error = ref<unknown | null>(null)
+
+  watch(device, () => {
+    connectToBluetoothGATTServer()
+  })
+
+  async function requestDevice(): Promise<void> {
+    // This is the function can only be called if Bluetooth API is supported:
+    if (!isSupported) return
+
+    // Reset any errors we currently have:
+    error.value = null
+
+    // If filters specified, we need to ensure we  don't accept all devices:
+    if (filters && filters.length > 0)
+      acceptAllDevices = false
+
+    try {
+      device.value = await navigator?.bluetooth.requestDevice({
+        acceptAllDevices,
+        filters,
+        optionalServices,
+      })
+    }
+    catch (err) {
+      error.value = err
+    }
+  }
+
+  const server = ref<undefined | BluetoothRemoteGATTServer>()
+
+  const isConnected = computed((): boolean => {
+    return server.value?.connected || false
+  })
+
+  async function connectToBluetoothGATTServer() {
+    // Reset any errors we currently have:
+    error.value = null
+
+    if (device.value && device.value.gatt) {
+      // Add callback to gattserverdisconnected event:
+      device.value.addEventListener('gattserverdisconnected', () => {})
+
+      try {
+        // Connect to the device:
+        server.value = await device.value.gatt.connect()
+      }
+      catch (err) {
+        error.value = err
+      }
+    }
+  }
+
+  tryOnMounted(() => {
+    if (device.value)
+      device.value.gatt?.disconnect()
```

Comment:
Disconnect on mounted? Is this intentional?

Reply:
Ahhh this should be connect()